sphereon-kmp-crypto-kms-azure/src/jsMain/kotlin/com/sphereon/crypto/kms/azure/AzureKeyVaultCryptoProvider.js.kt

```diff
@@ -0,0 +1,147 @@
+    interface CreateEcKeyOptions {
+        var curve: String?
+    }
+
+    class KeyClient(keyvaultUrl: String, credential: AzureIdentity.ClientSecretCredential) {
+        fun createEcKey(keyName: String, options: CreateEcKeyOptions): Promise<AzureKeyvaultKey>
+        fun getKey(keyName: String): Promise<AzureKeyvaultKey>
+    }
+
+    class CryptographyClient(key: AzureKeyvaultKey, credential: AzureIdentity.ClientSecretCredential) {
+        fun signData(
+            algorithm: String,
+            data: ByteArray
+        ): Promise<AzureKeyvaultSignDataResult>
+
+        fun verifyData(
+            algorithm: String,
+            data: ByteArray,
+            signature: ByteArray
+        ): Promise<AzureKeyvaultVerifyDataResult>
+    }
+}
+
+actual class AzureKeyvaultCryptoProvider actual constructor(
+    config: AzureKeyvaultClientConfig
+) : BaseAzureKeyvaultCryptoProvider(config.applicationId) {
+    private val keyClient: AzureKeyvaultKeys.KeyClient // Representing the Azure Key Vault client
+    private val clientSecretCredential: AzureIdentity.ClientSecretCredential
+
+    init {
+        if (config.credentialOpts.secretCredentialOpts == null) {
+            throw IllegalArgumentException("Azure Key Vault requires a secret credential")
+        }
+        val credential =
+            AzureIdentity.ClientSecretCredential(
+                config.tenantId,
+                config.credentialOpts.secretCredentialOpts.clientId,
+                config.credentialOpts.secretCredentialOpts.clientSecret
+            )
+        keyClient = AzureKeyvaultKeys.KeyClient(config.keyvaultUrl, credential)
+        clientSecretCredential = AzureIdentity.ClientSecretCredential(
+            config.tenantId,
+            config.credentialOpts.secretCredentialOpts.clientId,
+            config.credentialOpts.secretCredentialOpts.clientSecret
+        )
+    }
+
+    override suspend fun generateKeyAsync(
+        kmsKeyRef: String?,
+        use: JwkUse?,
+        keyOperations: Array<out KeyOperations>?,
+        alg: SignatureAlgorithm?
+    ): ManagedKeyPair {
+        val signatureAlgorithm = alg ?: SignatureAlgorithm.ECDSA_SHA256
+        if (!isSupportedSignatureAlgorithm(signatureAlgorithm)) {
+            throw IllegalArgumentException("Signature algorithm ${signatureAlgorithm.cryptoAlgorithm.name} is not supported by Azure Key Vault")
+        }
+        val keyName = kmsKeyRef ?: "key-${Uuid.v4String()}"
+        val options: AzureKeyvaultKeys.CreateEcKeyOptions = js("{}").unsafeCast<AzureKeyvaultKeys.CreateEcKeyOptions>().apply {
+            curve = signatureAlgorithm.curve?.jose?.value
+        }
+        val keyVaultKey = keyClient.createEcKey(keyName, options).await()
+        val keyVaultJwk = keyVaultKey.toJwk()
+        val publicCoseKey = CoseJoseKeyMappingService.toCoseKey(keyVaultJwk)
+        val kid = keyVaultKey.key.kid
+        return ManagedKeyPair(
+            kms = getId(),
+            kmsKeyRef = keyVaultKey.name,
```

Comment:
Is this guaranteed to be present and an actual thumbprint by default, or is it set to some other value?

Reply:
Its guaranteed to be present, as it is the first argument in the `createKey` call and if not passed, a random id is generated in the format `key-${Uuid.v4String()}`